From 6cc870c64a9fb31c02a62b18464b4464bf89dd8b Mon Sep 17 00:00:00 2001
From: Jonathan M Hsieh <jon@cloudera.com>
Date: Fri, 15 Jan 2016 16:09:41 -0800
Subject: [PATCH 290/295] CDH-34757 IntegrationTests time out due to
 RemoveColumnAction and chaos monkey cleanup

This patch is a small part of HBASE-11580 and included in Apache HBase 1.1.0+

Change-Id: I48d0f33ddfc5c7bb482e976460510ef41a25e8c5
Author: Jonathan Hsieh
Reason: Testing
Ref: CDH-34757
---
 .../hbase/chaos/actions/RemoveColumnAction.java    |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/RemoveColumnAction.java b/hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/RemoveColumnAction.java
index efb4413..c083d9c 100644
--- a/hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/RemoveColumnAction.java
+++ b/hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/RemoveColumnAction.java
@@ -54,7 +54,7 @@ public class RemoveColumnAction extends Action {
     HTableDescriptor tableDescriptor = admin.getTableDescriptor(tableName);
     HColumnDescriptor[] columnDescriptors = tableDescriptor.getColumnFamilies();
 
-    if (columnDescriptors.length <= 1) {
+    if (columnDescriptors.length <= (protectedColumns == null ? 1 : protectedColumns.size())) {
       return;
     }
 
-- 
1.7.9.5

